class MyException() extends Exception {
  fun getMessage(): String {
    ""
  }
}

@no_inline
fun throwExc(): String {
  throw MyException()
}

fun main(): void {
  print_raw(
    try {
      throwExc()
    } catch {
    | MyException() -> "Pass\n"
    | _ -> "Fail"
    }
  )
}
